design/specs/visual-query-builder.md

```diff
@@ -0,0 +1,140 @@
+
+To select a table, the user will find it on the list and click on the desired table option.
+
+Selecting the base table will enable the interface controls for adding additional steps, which were initially disabled. No action is available before a base table is selected.
+
+Once a selection has been made, the user will be prompted to proceed with the input table configuration, starting with column selection.
+
+[BASE TABLE SELECT WIREFRAME]
+
+## 2. Creating the Input Table
+
+The input table is created by selecting columns from the base table or creating new ones via formulas.
+
+Various options are available for input columns, including filters or aggregations that can help users control which data they want to show and how.
+
+These options can be set for all input columns, including formulas. Some options like aggregations might only be available for columns that link to multiple values. The system will automatically detect this and present the available options to users.
+
+### 2.1. Selecting Input Columns
+
+When selecting input columns, the user can add them using the list controls or drop the columns directly into the result table from the column selector.
+
+The column selector component will display all columns from the base table, as well as columns from any table linked or containing links to the base table.
+
+#### Direct Columns
+
+Direct columns are columns from the base or linked tables that contain the actual stored values, these can be selected directly and added as input columns.
+
+Any non-direct column from the base table that was set as a link in the base table will not be available for selection.
+
+[COLUMN SELECTOR WIREFRAME]
+
+#### Link Columns
+
+Link columns have foreign key constraints and hold the primary key values that link to other tables. These columns aren't selectable as direct columns. Instead, the tables that they link to are represented as expandable list groups. Inside each group, users can select the linked table columns. The join will determine the values.
+
+Link columns can exist in the base table or in other tables as links to the base table. However, the same pattern applies to any link column.
+
+The design should support multiple levels of nesting, depending on how the tables are linked.
+
+[COLUMN SELECTOR WIREFRAME]
+
+### 2.2 Adding Formulas
+
+During column selection, the user can add a new column by using formulas to generate its values. Available formulas will be included as part of the column selector, and dropping them into the result table will prompt users to set the formula settings for the selected formula.
+
+Depending on the selected formula, different settings will be available. Formulas will use the column selector component for settings that require the selection of columns. However, the system only list those with data types accepted by the formula.
+
+[WIREFRAME SHOWING FORMULA COLUMN]
```

Comment:
I think eventually we will need UX (wireframes) for all the different variable types supported by formulas.

Reply:
It can either be in this spec or a new spec.